lib/src/lib.rs

```diff
@@ -397,17 +398,28 @@ pub enum SessionResult {
 }
 
 // TODO: enrich with thiserror
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq, ThisError)]
```

Comment:
You could write directly `#[derive(thiserror::Error, PartialEq, Eq, Debug)]` which I found easier to understand than to check the structure and have to remind that is an alias.

Reply:
sure!